### PR TITLE
feat(collections): rewrite friends collections, friendspicker

### PR DESCRIPTION
**Pull request by hypeJunction**
BREAKING CHANGE
Friend collections have been rewritten and moved into their own plugin: friends_collections.
Friendspicker is now a slightly modified userpicker. All friendspicker CSS, JS and views have been removed.

collections1
collections2
collections3

This is also refactoring the access table, so:
- [x] Add/update tests
- [x] Update docs

### CODE REVIEW

File: mod/friend_collections/manifest.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	<version>3.0</version>
+	<category>bundled</category>
+	<category>friends</category>
+	<description>Provides an interface for users to manage their friend collections</description>
+	<website>http://www.elgg.org/</website>
+	<copyright>See COPYRIGHT.txt</copyright>
+	<license>GNU General Public License version 2</license>
+	<requires>
+		<type>elgg_release</type>
+		<version>2.2</version>
```
Comment (mrclay) [inline mod/friend_collections/manifest.xml]: Is this just to allow operation for now? Should be 3.0?

Comment (hypeJunction) [inline mod/friend_collections/manifest.xml]: Can't activate otherwise

**Comment by mrclay** [comment]
Need to update `.tx/config`. Kind of a defining moment: should `friend_collection` be a separate repo?

**Comment by hypeJunction** [comment]
I am fine either way. I feel this is quite close to core functionality so can be kept in, but would be a good exercise to move to aseparate repo

**Comment by mrclay** [comment]
> good exercise to move to a separate repo

My thinking as well. Where do docs live? Transifex setup? Would've been a good topic during hackathon. Need more sync communication.

**Comment by hypeJunction** [comment]
@mrclay I think for the sake of convenience we keep this in core for now and then detach with the rest of plugins.

**Comment by jdalsem** [comment]
This could probably be moved into the friends plugin now

**Comment by jdalsem** [comment]
this also fixes #9092

**Comment by hypeJunction** [comment]
I have managed to salvage this though there were a lot of conflicts. Let's review and merge ASAP before it grows stale on us again.
I have added test for most common operations. We can extend the scope later.

**Comment by jdalsem** [comment]
I see some elgg_view_input... should be elgg_view_field.... otherwise this can be merged when conflict has been resolved

**Comment by hypeJunction** [comment]
Rebased.
Replaced `elgg_view_input()` with `elgg_view_field()`

**Comment by hypeJunction** [comment]
Done.

**Pull request merged**